CMakeLists.txt

```diff
@@ -27,22 +27,45 @@ if (NOT CMAKE_INSTALL_PREFIX)
     set(CMAKE_INSTALL_PREFIX "./cmake_build")
 endif (NOT CMAKE_INSTALL_PREFIX)
 
-if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-    message(STATUS "==============OSX Environment================")
+OPTION (USE_OpenMP "Use OpenMP to enable <omp.h>" ON)
+
+# TODO: Distinguish Apple M1 chip from other Apple chips.
+IF (APPLE)
     set(MACOS TRUE)
-    set(CMAKE_CXX_COMPILER "/usr/local/opt/llvm/bin/clang++")
-    set(CMAKE_CXX_FLAGS "-I/usr/local/opt/llvm/include -I/usr/local/include -I/usr/local/opt/openblas/include -I/usr/local/opt/libomp/include")
-    check_cxx_compiler_flag(-std=c++11 HAS_STD_CPP11_FLAG)
-    if(HAS_STD_CPP11_FLAG)
-     add_compile_options(-std=c++11)
+    set(CMAKE_C_COMPILER "/opt/homebrew/Cellar/llvm/13.0.0_2/bin/clang")
+    set(CMAKE_CXX_COMPILER "/opt/homebrew/Cellar/llvm/13.0.0_2/bin/clang++")
```

Comment:
Do not use a specific path that may not work on other laptops.

Reply:
done.